packages/server/src/sdk/app/rows/search/sqs.ts

```diff
@@ -185,22 +195,45 @@ export async function search(
       }
     )
 
-    const output = {
-      rows: await outputProcessing<Row[]>(table, processed, {
-        preserveLinks: true,
-        squash: true,
-      }),
-    }
+    // get the rows
+    let finalRows = await outputProcessing<Row[]>(table, processed, {
+      preserveLinks: true,
+      squash: true,
+    })
 
+    // check if we need to pick specific rows out
     if (options.fields) {
       const fields = [...options.fields, ...CONSTANT_INTERNAL_ROW_COLS]
-      output.rows = output.rows.map((r: any) => pick(r, fields))
+      finalRows = finalRows.map((r: any) => pick(r, fields))
     }
 
-    return output
+    // check for pagination
+    if (paginate && limit) {
+      const response: SearchResponse<Row> = {
+        rows: finalRows,
+      }
+      const prevLimit = request.paginate!.limit
+      request.paginate = {
+        limit: 1,
+        page: bookmark * prevLimit + 1,
+      }
+      // check if there is another row
+      const nextRow = await runSqlQuery(request, allTables)
```

Comment:
I get you don't want to fuck with this too much, but it'd be better to initially query for `limit + 1` then discard the last row, and set `hasNextPage = true` when that happens.